users/src/main/java/com/familywarehouse/users/service/imp/UserServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.familywarehouse.users.service.imp;
+
+import com.familywarehouse.users.dto.UserDto;
+import com.familywarehouse.users.entity.User;
+import com.familywarehouse.users.exception.ResourceNotFoundException;
+import com.familywarehouse.users.exception.UserAlreadyExistsException;
+import com.familywarehouse.users.mapper.UserMapper;
+import com.familywarehouse.users.repository.UserRepository;
+import com.familywarehouse.users.service.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    //    private final PasswordEncoder passwordEncoder;
+    private final UserRepository userRepository;
+
+    @Override
+    public void createUser(UserDto userDto) {
+        String email = userDto.getEmail();
+        Optional<User> existedUser = userRepository.findByEmail(email);
+        if (existedUser.isPresent()) {
```

Comment:
1. Let's use Oprional API with Optional classes 
2. Can we rely on unique constraint to avoid additional select?